src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestForCliGit.java

```diff
@@ -0,0 +1,255 @@
+
+    @Rule
+    public GitClientSampleRepoRule secondRepo = new GitClientSampleRepoRule();
+
+    @Rule
+    public GitClientSampleRepoRule thirdRepo = new GitClientSampleRepoRule();
+
+    private int logCount = 0;
+    private final Random random = new Random();
+    private static final String LOGGING_STARTED = "Logging started";
+    private LogHandler handler = null;
+    private TaskListener listener;
+    private final String gitImplName;
+
+    private String revParseBranchName = null;
+
+    private int checkoutTimeout = -1;
+    private int cloneTimeout = -1;
+    private int fetchTimeout = -1;
+    private int submoduleUpdateTimeout = -1;
+
+
+    WorkspaceWithRepo workspace;
+
+    private GitClient testGitClient;
+    private File testGitDir;
+    private CliGitCommand cliGitCommand;
+
+    public GitAPITestForCliGit(final String gitImplName) {
+        this.gitImplName = gitImplName;
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection gitObjects() {
+        List<Object[]> arguments = new ArrayList<>();
+        String[] gitImplNames = {"git"};
+        for (String gitImplName : gitImplNames) {
+            Object[] item = {gitImplName};
+            arguments.add(item);
+        }
+        return arguments;
+    }
+
+    @BeforeClass
+    public static void loadLocalMirror() throws Exception {
+        /* Prime the local mirror cache before other tests run */
+        /* Allow 2-5 second delay before priming the cache */
+        /* Allow other tests a better chance to prime the cache */
+        /* 2-5 second delay is small compared to execution time of this test */
+        Random random = new Random();
+        Thread.sleep((2 + random.nextInt(4)) * 1000L); // Wait 2-5 seconds before priming the cache
+        TaskListener mirrorListener = StreamTaskListener.fromStdout();
+        File tempDir = Files.createTempDirectory("PrimeGITAPITest").toFile();
+        WorkspaceWithRepo cache = new WorkspaceWithRepo(tempDir, "git", mirrorListener);
+        cache.localMirror();
+        Util.deleteRecursive(tempDir);
+    }
+
+    @Before
+    public void setUpRepositories() throws Exception {
+        revParseBranchName = null;
+        checkoutTimeout = -1;
+        cloneTimeout = -1;
+        fetchTimeout = -1;
+        submoduleUpdateTimeout = -1;
+
+        Logger logger = Logger.getLogger(this.getClass().getPackage().getName() + "-" + logCount++);
+        handler = new LogHandler();
+        handler.setLevel(Level.ALL);
+        logger.setUseParentHandlers(false);
+        logger.addHandler(handler);
+        logger.setLevel(Level.ALL);
+        listener = new hudson.util.LogTaskListener(logger, Level.ALL);
+        listener.getLogger().println(LOGGING_STARTED);
+
+//        workspace = new WorkspaceWithRepo(repoRoot, gitImplName, listener);
+//        workspace = new WorkspaceWithRepo(temporaryDirectoryAllocator.allocate(), gitImplName, listener);
```

Comment:
Since repoRoot has been removed above, it would be good to remove it here as well.
```suggestion
```

Reply:
Yes, you're actually I missed out this one when removing. Will complete it, but need some time.